src/runtime_src/core/pcie/driver/linux/xocl/subdev/icap.c

```diff
@@ -2516,12 +2516,13 @@ static int __icap_download_bitstream_user(struct platform_device *pdev,
 	xdev_handle_t xdev = xocl_get_xdev(pdev);
 	int err = 0;
 
-	/* TODO: Use slot handle to unregister CUs. CU subdev will be destroyed */
-	xocl_subdev_destroy_by_level(xdev, XOCL_SUBDEV_LEVEL_URP);
+	/* Use slot handle to unregister CUs. CU subdev will be destroyed */
 	err = xocl_unregister_cus(xdev, slot_id);
 	if (err && (err != -ENODEV))
 		goto done;
 
+	xocl_subdev_destroy_by_level_skip_cus(xdev, XOCL_SUBDEV_LEVEL_URP);
```

Comment:
Could you associate subdev with xclbin? E.g. add a xclbin(slot) id into subdev? Thus you may add interface xocl_subdev_destroy_by_xclbin

Reply:
Yes. I have done this. Used slot instead of xclbin